airflow/www/static/js/components/RenderedJsonField.tsx

```diff
@@ -47,7 +47,7 @@ const JsonParse = (content: string | object) => {
       contentJson = content;
     }
     contentFormatted = JSON.stringify(contentJson, null, 4);
-    isJson = true;
+    isJson = typeof contentJson === "object"; // ensure numbers/bool are not treated as JSON
```

Comment:
This also excludes `null`. I’m not even sure if it’s a possible value here, or how we are expected to treat it.

Reply:
Oh. You are really good. Did not have "NULL" in my test set as in all my examples NULL was never persisted in XCom, in these cases just no XCom was stored. But actually if I try we have even a second problem with NULL in the XCom value.

![image](https://github.com/user-attachments/assets/401f739d-f0df-4d13-b73b-89f714954999)

Reply:
Okay, got a bit more complex now but extracted the utility function and added a test for it.